source/WindowScroller/WindowScroller.js

```diff
@@ -138,25 +136,24 @@ export default class WindowScroller extends React.PureComponent<Props, State> {
     this._isMounted = true;
   }
 
-  componentWillReceiveProps(nextProps: Props) {
-    const scrollElement = this.props.scrollElement;
-    const nextScrollElement = nextProps.scrollElement;
+  componentDidUpdate(prevProps: Props, prevState: State) {
+    const {scrollElement} = this.props;
 
     if (
-      scrollElement !== nextScrollElement &&
-      scrollElement &&
-      nextScrollElement
+      prevProps.scrollElement !== scrollElement &&
+      prevProps.scrollElement != null &&
+      scrollElement != null
     ) {
-      this.updatePosition(nextScrollElement, nextProps);
+      this.updatePosition(scrollElement);
 
-      unregisterScrollListener(this, scrollElement);
-      registerScrollListener(this, nextScrollElement);
+      // $FlowFixMe Why is this necessary? There's a check right above.
+      unregisterScrollListener(this, prevProps.scrollElement);
```

Comment:
because updatePosition may change this value value.

Reply:
Ah! Nice observation.

Reply:
before is worked ok, because it was const instead of field